src/components/BoardsColumn.vue

```diff
@@ -0,0 +1,134 @@
+    <div class="flex gap-6 h-max">
+      <div v-for="column in columns" :key="column.name" class="flex flex-col">
+        <div class="flex items-center gap-2 mb-8">
+          <div
+            class="h-[15px] w-[15px] rounded-full"
+            :class="circleColor(column)"
+          ></div>
+          <p class="text-xs text-gray-400 uppercase">
+            {{ column.name }} ({{ column.tasks.length }})
+          </p>
+        </div>
+        <task-card
+          v-for="{ title, subtasks } in column.tasks"
+          :key="title"
+          :howManyCompleted="returnNumberOfCompletedSubtasks(subtasks)"
+          :howManySubtasks="subtasks.length"
+          :title="title"
+          :isClickedTask="clickedTitle === title"
+          :showTaskDetails="(e: Event) => changeClickedTask(e)"
+        />
+      </div>
+      <div class="new-column group" tabindex="0">
+        <span class="new-column-text">+ New Column</span>
+      </div>
+    </div>
+    <Teleport to="body">
+      <transition name="dialog">
+        <see-task-dialog
+          v-if="clickedTitle != null"
+          :showEditForm="showEditForm"
+          :showDeleteForm="showDeleteForm"
+          :closeSeeTask="closeSeeTask"
+        />
+      </transition>
+    </Teleport>
+    <transition name="dialog">
+      <confirmation-dialog
+        v-if="isDeleteTaskDialogShown"
+        elementToDelete="task"
+        elementName="Research pricing points of various competitors and trial different business models"
+        :closeDialog="() => (isDeleteTaskDialogShown = false)"
+      />
+    </transition>
+    <transition name="dialog">
+      <task-dialog
+        v-if="isEditTaskDialogShown"
+        action="edit"
+        :selectedMultiOptionItems="selectedMultiOptionItems"
+        :closeDialog="() => (isEditTaskDialogShown = false)"
+      />
+    </transition>
+  </div>
+</template>
+
+<script setup lang="ts">
+import type { BoardColumn, Task } from '../api/boardsTypes'
+import TaskCard from './TaskCard.vue'
+import SeeTaskDialog from './Dialogs/SeeTaskDialog.vue'
+import ConfirmationDialog from '../components/Dialogs/ConfirmationDialog.vue'
+import TaskDialog from '../components/Dialogs/TaskDialog.vue'
+import { returnBoardProperties } from '../composables/boardProperties'
+import { returnNumberOfCompletedSubtasks } from '../composables/completedTasks'
+import { computed, ref, Ref } from 'vue'
+
+const props = defineProps<{
+  columns: BoardColumn[]
+  logo: boolean
+}>()
+
+const circleColor = computed(() => {
+  return (column: BoardColumn) => ({
+    'bg-blue-600': props.columns.indexOf(column) % 3 === 0,
+    'bg-blue-500': props.columns.indexOf(column) % 3 === 1,
+    'bg-green-400': props.columns.indexOf(column) % 3 === 2
+  })
+})
+
+const clickedTitle: Ref<null | Task['title']> = ref(null)
+const changeClickedTask = (e: Event) => {
+  const titleOfClickedTask = (e.currentTarget as HTMLElement)?.querySelector(
+    'p'
+  )?.textContent
```

Comment:
I guess, it's done but I'm not really sure about my solution being right. 

In my child component, I have the code like this:
```js
<template>
  <div
    @click="$emit('changeTaskTitle', title)"
    tabindex="0"
    class="task-card group"
  >
    <p class="task-card__title">
      {{ title }}
    </p>
    <p class="text-gray-400 text-xs">
      {{ howManyCompleted }} of {{ howManySubtasks }} subtasks
    </p>
  </div>
</template>

<script setup lang="ts">
defineProps<{
  title: string
  howManyCompleted: number
  howManySubtasks: number
}>()
defineEmits(['changeTaskTitle'])
</script>
```

but according to the Vue docs ([https://vuejs.org/guide/components/events.html#event-arguments](here)), I don't really need this `defineEmits` thing. I added it because eslint yelled at me I should do so, but I don't think I understand their exaplanation [https://eslint.vuejs.org/rules/require-explicit-emits.html](here) on why this is the correct approach.

Also, I'd like to emphasize here, this part of the code will be changed in the future anyway, because, currently, all related data is hardcoded.

Reply:
Why not just `change`?

Reply:
Because, recently, I've liked this idea of naming things (functions, variables) the most descriptive way possible. If I was to come back to this code in a few weeks, let say, I would have no idea what `change` means and what's supposed to be changed. Whereas, `changeTaskTitle` seems to be pretty self-descriptive to me.

And I believe we both know it's highly possible I will be coming back to this code many times in the not-so-distant future, so I'd better know what my functions do.

Reply:
Yeah... so, the whole function that edits the task title looks like this:
```js
@change-task-title="(title) => (clickedTitle = title)"
```

So, if I wanted to change `description` as well, I would have to repeat myself anyway.

I wasn't familiar with the KISS rule but "simple" to me would mean descriptive enough so I know what it does without reading the whole code.

So, not really conviced on this one, but since you insist - I **changed it**.